doc/whats_new/v1.1.rst

```diff
@@ -119,6 +119,9 @@ Changelog
   get accurate results when the number of features is large.
   :pr:`21109` by :user:`Smile <x-shadow-man>`.
 
+- |Enhancement| `dict_learning` and `dict_learning_online` methods preserve dtype for numpy.float32.
+  :pr:`22002` by :user:`Takeshi Oura <takoika>`.
+
```

Comment:
Your changes also impact `DictionaryLearning`, `MiniBatchDictionaryLearning` and `SparseCoder`. Please mention them here. I think it would also be nice to add similar tests for those as well

Reply:
Changed.
Also unit tests to verify dtype matching for `DictionaryLearning`,` MiniBatchDictionaryLearning` and `SparseCoder` are added. Then a test for numerical consistency among `np.float32` and `np.float64` in `sparse_encode` method is added.